plio/io/io_controlnetwork.py

```diff
@@ -29,6 +32,71 @@ def write_filelist(lst, path="fromlist.lis"):
         handle.write('\n')
     return
 
+
+class MessageType(IntEnum):
+    """
+    An enum to mirror the ISIS3 MeasureLogData enum.
+    """
+    GoodnessOfFit = 2
+    MinimumPixelZScore = 3
+    MaximumPixelZScore = 4
+    PixelShift = 5
+    WholePixelCorrelation = 6
+    SubPixelCorrelation = 7 
+
+class Log():
+    
+    def __init__(self, messagetype, value):
+        """
+        A protobuf compliant log object.
+        
+        Parameters
+        ----------
+        messagetype : int or str
+                      Either the integer or string representation from the MessageType enum
+                      
+        value : int or float
+                The value to be stored in the message log
+        """
+        if isinstance(messagetype, int):
+            # by value
+            self.messagetype = MessageType(messagetype)
+        else:
+            # by name
+            print(MessageType[messagetype])
+            self.messagetype = MessageType[messagetype]
+        
+        if not isinstance(value, (float, int)):
+            raise TypeError(f'{value} is not a numeric type')
+        self.value = value
+        
+    def __repr__(self):
+        return f'{self.messagetype.name}: {self.value}'
+        
+    def to_protobuf(self):
+        """
+        Return protobuf compliant measure log object representation
+        of this class.
+        
+        Returns
+        -------
+        log_message : obj
+                      MeasureLogData object suitable to append to a MeasureLog
+                      repeated field.
+        """
+        # I do not see a better way to get to the inner MeasureLogData obj than this
+        # imports were not working because it looks like these need to instantiate off
+        # an object
+        log_message = cnf.ControlPointFileEntryV0002().Measure().MeasureLogData()
+        log_message.doubleDataValue = self.value
+        log_message.doubleDataType = self.messagetype
```

Comment:
MeasureLogData also appears in ControlPointFileEntryV0005_pb2.py (cnp5), should we do a version check here?

Reply:
The controlnetwork to_isis code takes an optional version parameter. I think adding that here would be good.

Reply:
I added it here, but the write only writes out a version 2, so that might be a nice enhancement at some point to write a v5.